zoobc/zoobc-core
# 1424 db tx priority locks

## Description
Since there is no way to prioritize database transactions at the moment, in case the node is overwhelmed by large number of transactions or other concurrent processes that lead to db writes, important and lenghtly processes such as processblock, processfork and downloadblockchain could be blocked by the first ones.

This is due to the fact that every time we execute a write operation or begin a db transaction, we lock other similar processes until the query has finished.

To avoid many low priority queries (such as writing new mempool transactions) to starve and block more important ones, we need to implement a priority sytem for database transactions.

## Breakdown
- [x] Added this priorityLock implementation to our codebase (from https://github.com/platinummonkey/priority-lock)
- [x] Refactored the relevant code (query/executor) to use priority locks instead of regular mutex ones

## Reference Issue
Close #1424 

## Step to Test (optional)
- deploy on alpha network
- spam with at least 10tps (with and without antispam filter)
- verify that no nodes fork
